test/puppetlabs/ssl_utils/core_test.clj

```diff
@@ -550,6 +550,22 @@
                  #"The CRL reader does not contain a CRL matching.*"
                  (pem->ca-crl crl-chain leaf-cert)))))))
 
+    (testing "verify CRL against cert chain"
+      (let [leaf-name-string "CN=Intermediate CA (master-ca)"
+            leaf-name (cn leaf-name-string)
+            leaf-keypair (generate-key-pair)
+            leaf-public-key (get-public-key leaf-keypair)
+            serial 42
+            not-before (generate-not-before-date)
+            not-after (generate-not-after-date)
+            leaf-cert (sign-certificate issuer-name private-key serial not-before
+                                        not-after leaf-name leaf-public-key
+                                        (create-ca-extensions leaf-name serial leaf-public-key))]
+        (testing "when CRL was issued by cert in chain"
+          (is (true? (crl-issued-by-cert-in-chain? crl [leaf-cert cert]))))
+        (testing "when CRL was not issued by cert in chain"
+          (is (false? (crl-issued-by-cert-in-chain? crl [leaf-cert]))))))
+
```

Comment:
wasn't sure how useful this test is either. the setup is potentially unnecessary, but figured it made more sense to test with certs in the chain, rather than an empty chain.

Reply:
Agreed. I think it's good to have.